From 6f998b72dd99c36000bbf806c128edf97bda0574 Mon Sep 17 00:00:00 2001
From: Roland Haas <roland.haas@physics.gatech.edu>
Date: Wed, 19 Oct 2011 06:22:04 -0700
Subject: [PATCH] use diff-files instead of status to decide when to stash
 this works with git 1.6 which does not yet support --porcelain

---
 GetComponents |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/GetComponents b/GetComponents
index 85128c5..e21b33b 100755
--- a/GetComponents
+++ b/GetComponents
@@ -1619,8 +1619,8 @@ sub git_stash_update_repo {
     my ( $git_repo, $repo_loc, $checkout, @branches ) = @_;
     
     # determine if we need to stash local changes
-    my ( $err, $out ) = run_command("cd $repo_loc && $git status --porcelain --untracked-files=no");
-    my $stash = $out;
+    my ( $err, $out ) = run_command("cd $repo_loc && $git diff-files --quiet");
+    my $stash = $err;
     if ($stash) {
         # found local changes, stash them
         my ( $err, $out ) = run_command("cd $repo_loc && $git stash");
-- 
1.7.1

